### PR TITLE
fix another breaking `typer` change

### PR DESCRIPTION
**Pull request by zzstoatzz**
I'll leave it up for discussion whether we should upper bound pin the minor version, this is the second time they've released a minor version recently that's broken our CLI

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,4 +15,4 @@ kubernetes >= 24.2.0, < 30.0.0
 pytz >= 2021.1, < 2025
 readchar >= 4.0.0, < 5.0.0
 sqlalchemy[asyncio] >= 1.4.22, != 1.4.33, < 3.0.0
-typer >= 0.4.2
+typer >= 0.12.0
```
Comment (zzstoatzz) [inline requirements.txt:18]: 👍  👎  ?
```suggestion
typer >= 0.12.0< 0.13.0
```

Comment (bunchesofdonald) [reply requirements.txt:18]: I'd vote 👎 , upper bounds tend to be pretty annoying for downstream users. They're fine if we have a known incompatibility, but I wouldn't, in general, favor pre-emptive pinning.

**Review by bunchesofdonald — APPROVED** [review]

**Comment by d1manson** [comment]
I'm getting an error now which seems to be related to this:

```
Traceback (most recent call last):
  File "/home/circleci/repo/venv/bin/prefect", line 5, in <module>
    from prefect.cli import app
  File "/home/circleci/repo/venv/lib/python3.9/site-packages/prefect/cli/__init__.py", line 3, in <module>
    from prefect.cli.root import app
  File "/home/circleci/repo/venv/lib/python3.9/site-packages/prefect/cli/root.py", line 17, in <module>
    from prefect.cli._types import PrefectTyper, SettingsOption
  File "/home/circleci/repo/venv/lib/python3.9/site-packages/prefect/cli/_types.py", line 17, in <module>
    def SettingsOption(setting: Setting, *args, **kwargs) -> typer.Option:
AttributeError: module 'typer' has no attribute 'Option'
```

**Comment by zzstoatzz** [comment]
hi @d1manson - what version are you on? the newest release (in the last hour) should fix this

**Comment by d1manson** [comment]
> hi @d1manson - what version are you on? the newest release (in the last hour) should fix this

ahh yes, i'm on a slightly older version. thanks

**Pull request merged**